client/src/pages/admin/MergeLocations.js

```diff
@@ -0,0 +1,516 @@
+            reversed={reversed}
+            action={getActionButton(() => {
+              setFieldValue("lat", location.lat)
+              setFieldValue("lng", location.lng)
+            })}
+          />
+          {map}
+          <div className={"merge-loc-fset" + (reversed ? " reversed" : "")}>
+            <ApprovalSteps
+              type={PLANNING_APPROVAL}
+              steps={location.planningApprovalSteps}
+              fieldSetAction={getActionButton(() =>
+                setFieldValue(
+                  "planningApprovalSteps",
+                  location.planningApprovalSteps
+                )
+              )}
+            />
+          </div>
+
+          <div className={"merge-loc-fset" + (reversed ? " reversed" : "")}>
+            <ApprovalSteps
+              type={PUBLICATION_APPROVAL}
+              steps={location.approvalSteps}
+              fieldSetAction={getActionButton(() =>
+                setFieldValue("approvalSteps", location.approvalSteps)
+              )}
+            />
+          </div>
+        </>
+      )}
+    </div>
+  )
+
+  function getActionButton(onClickAction) {
+    return (
+      <small>
+        <Button
+          icon={reversed ? "double-chevron-left" : ""}
+          rightIcon={reversed ? "" : "double-chevron-right"}
+          intent="primary"
+          text="Use value"
+          onClick={onClickAction}
+        />
+      </small>
+    )
+  }
+}
+
+LocationColumn.propTypes = {
+  location: PropTypes.instanceOf(Location),
+  setLocation: PropTypes.func.isRequired,
+  setFieldValue: PropTypes.func.isRequired,
+  reversed: PropTypes.bool,
+  style: PropTypes.object
+}
+
+LocationColumn.defaultProps = {
+  reversed: false
+}
+
+const LocationField = ({ label, value, align, reversed, action }) => {
+  return (
+    <div
+      style={{
+        borderBottom: "1px solid #CCCCCC",
+        display: "flex",
+        flexDirection: reversed ? "row-reverse" : "row",
+        justifyContent: "space-between",
+        alignItems: "center",
+        padding: "8px 0"
+      }}
+    >
+      <div style={{ flex: "1 1 auto" }}>
+        <div
+          style={{
+            fontWeight: "bold",
+            textAlign: align || "center",
+            textDecoration: "underline"
+          }}
+        >
+          {label}
+        </div>
+        <div style={{ textAlign: align || "center" }}>{value}</div>
+      </div>
+      {action}
+    </div>
+  )
+}
+
+LocationField.propTypes = {
+  label: PropTypes.string.isRequired,
+  value: PropTypes.node,
+  align: PropTypes.string,
+  reversed: PropTypes.bool,
+  action: PropTypes.node
+}
+
+const MergeLocations = ({ pageDispatchers }) => {
+  const history = useHistory()
+  const [saveError, setSaveError] = useState(null)
+  const [mergedLocation, setMergedLocation] = useState(new Location())
+  const [location01, setLocation01] = useState()
+  const [location02, setLocation02] = useState()
+
+  const setLocationFn01 = value => {
```

Comment:
I don't understand why these 2 functions are needed. They seem like a complicated workaround. What problem are we trying to solve?

Reply:
These 2 functions primarily exist for lifting state up. `MergeLocations` component has two children `LocationColumn` components. In each of these `LocationColumn` components we select `location01` and `location02`. We need `location01` and `location02` on parent component for following reasons:
 - to  prevent user from selecting same locations. (to show "Please select a different location!" warning)
 - to show "Please select both locations to proceed..." warning
 - to disable submit button accordingly
 - to reset mergeLocation when a new location is selected for either of locations